From 714739e69ead1d9823233af40645277f6d4633ea Mon Sep 17 00:00:00 2001
From: Peter Korsgaard <peter@korsgaard.com>
Date: Tue, 2 May 2017 23:21:46 +0200
Subject: [PATCH] cilk: fix build without wchar

When building against uClibc with wchar support disabled, WCHAR_MIN and
WCHAR_MAX are not defined leading to compilation errors.

Fix it by only including the wchar code if available.

Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
[Romain: convert to git patch]
Signed-off-by: Romain Naour <romain.naour@gmail.com>
---
 libcilkrts/include/cilk/reducer_min_max.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libcilkrts/include/cilk/reducer_min_max.h b/libcilkrts/include/cilk/reducer_min_max.h
index 641aa82..4f8e010 100644
--- a/libcilkrts/include/cilk/reducer_min_max.h
+++ b/libcilkrts/include/cilk/reducer_min_max.h
@@ -3289,7 +3289,9 @@ __CILKRTS_BEGIN_EXTERN_C
 CILK_C_REDUCER_MAX_INSTANCE(char,               char,       CHAR_MIN)
 CILK_C_REDUCER_MAX_INSTANCE(unsigned char,      uchar,      0)
 CILK_C_REDUCER_MAX_INSTANCE(signed char,        schar,      SCHAR_MIN)
+#ifdef WCHAR_MIN
 CILK_C_REDUCER_MAX_INSTANCE(wchar_t,            wchar_t,    WCHAR_MIN)
+#endif
 CILK_C_REDUCER_MAX_INSTANCE(short,              short,      SHRT_MIN)
 CILK_C_REDUCER_MAX_INSTANCE(unsigned short,     ushort,     0)
 CILK_C_REDUCER_MAX_INSTANCE(int,                int,        INT_MIN)
@@ -3441,7 +3443,9 @@ __CILKRTS_BEGIN_EXTERN_C
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(char,               char,       CHAR_MIN)
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(unsigned char,      uchar,      0)
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(signed char,        schar,      SCHAR_MIN)
+#ifdef WCHAR_MIN
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(wchar_t,            wchar_t,    WCHAR_MIN)
+#endif
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(short,              short,      SHRT_MIN)
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(unsigned short,     ushort,     0)
 CILK_C_REDUCER_MAX_INDEX_INSTANCE(int,                int,        INT_MIN)
@@ -3567,7 +3571,9 @@ __CILKRTS_BEGIN_EXTERN_C
 CILK_C_REDUCER_MIN_INSTANCE(char,               char,       CHAR_MAX)
 CILK_C_REDUCER_MIN_INSTANCE(unsigned char,      uchar,      CHAR_MAX)
 CILK_C_REDUCER_MIN_INSTANCE(signed char,        schar,      SCHAR_MAX)
+#ifdef WCHAR_MAX
 CILK_C_REDUCER_MIN_INSTANCE(wchar_t,            wchar_t,    WCHAR_MAX)
+#endif
 CILK_C_REDUCER_MIN_INSTANCE(short,              short,      SHRT_MAX)
 CILK_C_REDUCER_MIN_INSTANCE(unsigned short,     ushort,     USHRT_MAX)
 CILK_C_REDUCER_MIN_INSTANCE(int,                int,        INT_MAX)
@@ -3719,7 +3725,9 @@ __CILKRTS_BEGIN_EXTERN_C
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(char,               char,       CHAR_MAX)
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(unsigned char,      uchar,      CHAR_MAX)
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(signed char,        schar,      SCHAR_MAX)
+#ifdef WCHAR_MAX
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(wchar_t,            wchar_t,    WCHAR_MAX)
+#endif
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(short,              short,      SHRT_MAX)
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(unsigned short,     ushort,     USHRT_MAX)
 CILK_C_REDUCER_MIN_INDEX_INSTANCE(int,                int,        INT_MAX)
-- 
2.9.3

